autoscale/src/main/java/com/sequenceiq/periscope/service/UsageReportingService.java

```diff
@@ -0,0 +1,92 @@
+package com.sequenceiq.periscope.service;
+
+
+import java.util.Set;
+import java.util.stream.Stream;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+import com.cloudera.thunderhead.service.common.usage.UsageProto;
+import com.sequenceiq.cloudbreak.usage.UsageReporter;
+import com.sequenceiq.periscope.api.model.AlertType;
+import com.sequenceiq.periscope.api.model.ScalingStatus;
+import com.sequenceiq.periscope.domain.BaseAlert;
+import com.sequenceiq.periscope.domain.Cluster;
+
+@Service
+public class UsageReportingService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UsageReportingService.class);
+
+    @Inject
+    private UsageReporter usageReporter;
+
+    @Async
+    public void reportAutoscalingTriggered(int adjustmentCount, int clusterSize, ScalingStatus scalingStatus, String scalingAction,
+            BaseAlert alert, Cluster cluster) {
+        UsageProto.CDPDatahubAutoscaleTriggered.Builder cdpAutoscaleTriggerDetailsBuilder =
+                UsageProto.CDPDatahubAutoscaleTriggered.newBuilder().setAutoscaleTriggerDetails(
+                        UsageProto.CDPAutoscaleTriggerDetails.newBuilder()
+                                .setAccountId(cluster.getClusterPertain().getTenant())
+                                .setClusterCrn(cluster.getStackCrn())
+                                .setClusterName(cluster.getStackName())
+                                .setAutoscalingAction(scalingAction)
+                                .setAutoscaleStatus(getAutoscaleStatusType(scalingStatus))
+                                .setAutoscalingPolicyDefinition(withAlert(alert))
+                                .setOriginalHostGroupNodeCount(clusterSize)
+                                .setDesiredHostGroupNodeCount(clusterSize + adjustmentCount)
```

Comment:
The setter methods of UsageProto builder fail if a null value is passed to them so if you are not sure that these values cannot be null then it's advised to do a null check or set defaults (for example: .setClusterName(defaultIfEmpty(cluster.getStackName(), "")) )

Reply:
ok changed.